### PR TITLE
FEAT: Build CJS + ESM

### PR DESCRIPTION
**Pull request by ynotdraw**
⚠️  **NOTE:** Going into feature-branch, not master! ⚠️ 

## Changes
Attempts to address https://github.com/lifeomic/chromicons/issues/3. 

### Usage in App
I have an app where I'm `yarn link`ed and here's how my imports look:

```js
import { Apple, Activity } from '@lifeomic/chromicons';
```

And then they're used like standard React components:

```jsx
<Apple className="text-red-500" />
<Activity className="text-yellow-300" />
```

### Improvements
- I'd like to import these from `@lifeomic/chromicons/lined` specifically.  I'll look at that.  I talked to Doryan though, and for now it sounds like we will be focusing on lined icons. 
- Probably need a `package.json` in the `dist` directory; however, the way it works now is it gives you the entire source (maybe not a huge deal...?)

#### Importing SVGs Directly
With these changes, I believe SVGs can be imported directly via `import activitySVG from '@lifeomic/chromicons/src/raw/lined/activity.svg';`

---
#### Use `tsdx` for Builds

https://tsdx.io/ is a zero-config CLI for TS package development.  It makes publishing packages a lot easier, allowing for output of CJS, ESM, and/or UMD.

#### Migrate Raw SVGs
I moved all raw SVGs from `src/lined` to `src/raw/lined`.  More on this in a bit

#### `data-icon`
Add `data-icon` attributes to all SVGS via `svgr --template`.  We do this for `chroma-react` already, as the documentation team needs a way to figure out which icon is being used without having to scroll through pages and pages of icons.

The nice thing about doing this via a template, is if we decide we need to modify the root element more, we can do so via the `svg-template.js` file.

#### Automatically Generate React Components
SVGR can convert an svg file into a React component, so we leverage that (+ TS!) via `generate:react-components`.

#### Generate an Index File for `tsdx`
`tsdx` needs an `index.tsx` file in the `src` directory, so we use a script to generate that ( `generate:index-file` ).

**This means 0 React components are actually checked-in.** We rely on the build step below to do everything for us, but since it's all generated, no need to actually commit React components.

#### Build
Following all of the above, we can run `yarn build` and the dist is generated for us.  Here's a screenshot of the dist directory (and the index.js).  The React components live underneath `dist/lined`.

You can see from the `release.yml` file the steps you need to take to be able to use this package:

```bash
yarn
yarn generate:build-output
yarn build
```

Screen Shot 2020-11-20 at 2 44 49 PM

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    // interop between ESM and CJS modules. Recommended by TS
+    "esModuleInterop": true,
+    // significant perf increase by skipping checking .d.ts files, particularly those in node_modules. Recommended by TS
+    "skipLibCheck": true,
+    // error out if import and file system have a casing mismatch. Recommended by TS
+    "forceConsistentCasingInFileNames": true,
+    // `tsdx build` ignores this option, but it is commonly used when type-checking separately with `tsc`
+    "noEmit": true,
+  }
+}
```
Comment (ynotdraw) [inline tsconfig.json:35]: This was all done auto-magically with `tsdx`.  Seems fine to me though 👍

File: svg-template.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    },
+  ];
+  return template.ast`
+    ${imports}
+    const ${componentName} = (${props}) => ${jsx}
+    ${exports}
+  `;
+}
+
+module.exports = template;
```
Comment (ynotdraw) [inline svg-template.js:26]: More docs for this here: https://react-svgr.com/docs/custom-templates/

**Review by dexterca — APPROVED** [review]

File: svg-template.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+function template(
+  { template },
+  opts,
+  { imports, componentName, props, jsx, exports }
+) {
+  jsx.openingElement.attributes = [
+    ...jsx.openingElement.attributes,
+    {
+      type: 'JSXAttribute',
+      name: { type: 'JSXIdentifier', name: 'data-icon' },
```
Comment (travomic) [inline svg-template.js:15]: 😎

**Review by travomic — APPROVED** [review]

File: svg-template.js (hunk 3 above)
Comment (ynotdraw) [reply svg-template.js:15]: SVGR is awesome - super easy!

**Comment by ynotdraw** [comment]
> With these changes, I believe SVGs can be imported directly via import activitySVG from '@lifeomic/chromicons/src/raw/lined/activity.svg';

I actually have an idea. I think if I move the raw icons out of `src/` and into `lined/` the import will be something like:

` import activitySVG from '@lifeomic/chromicons/lined/activity.svg';`

Since this already has some approvals, and is in a feature branch, I'll save those changes for a downstream PR though after giving it a shot 👍

**Review by hemp — APPROVED** [review]

**Review by mschroering — APPROVED** [review]

**Pull request merged**